lualib/lua_scanners/icap.lua

```diff
@@ -240,10 +240,21 @@ local function icap_check(task, content, digest, rule, maybe_part)
 
       local function get_req_headers()
 
+        local in_filename, in_type, in_subtype, in_client_ip = nil
+        local parts = task:get_parts()
+        for _, part in pairs(parts) do
```

Comment:
I don't understand this logic. There could be multiple parts in a message with different filenames and content types. With this approach we just select the first one.

Reply:
I found out, that icap_check is called multiple times for each attachment passed as content. Because task variable is always the same and returns all the parts, and content variable doesn't have get_detected_type() method. So, on each call I get all the parts from task and check against content variable to see which one is actually the current one. 
As you wrote, now I can imagine how expensive get_parts can be, especially with larger attachments. 
I will try the "maybe_part" route as it seems the most logical one.
Just a few questions - if maybe_part is actually a mime part object, will it have the get_detected_type() and get_filename() methods? and as content is coming as a userdata without any methods can it be converted to mime part object?

Thanks,
Lyuben